dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/HttpUtils.java

```diff
@@ -467,20 +466,90 @@ public static String getProtocolHeader(Message m, String name, String defaultVal
 
     public static String getBaseAddress(Message m) {
         String endpointAddress = getEndpointAddress(m);
-        try {
-            URI uri = new URI(endpointAddress);
-            String path = uri.getRawPath();
-            String scheme = uri.getScheme();
+        //Liberty code change start
+        String[] addr = parse(endpointAddress, false);
+        if (addr != null) {
+            String scheme = addr[0];
+            String path = addr[1];
             if (scheme != null && !scheme.startsWith(HttpUtils.HTTP_SCHEME)
-                && HttpUtils.isHttpRequest(m)) {
+                    && HttpUtils.isHttpRequest(m)) {
                 path = HttpUtils.toAbsoluteUri(path, m).getRawPath();
             }
             return (path == null || path.length() == 0) ? "/" : path;
-        } catch (URISyntaxException ex) {
+        } else {
             return endpointAddress == null ? "/" : endpointAddress;
         }
     }
 
+    public static String[] parse(String addr, boolean parseAuthority) {
+        String scheme = null;
+        String parsedAuthorityOrRawPath = null;
+        int n = addr.length();
+        int p = scan(addr, 0, n, "/?#", ":");
+        if ((p >= 0) && at(addr, p, n, ':')) {
+            if (p == 0) {
+                return null;
+            }    
+            scheme = addr.substring(0, p);
+            p++;
+                if (at(addr, p, n, '/')) {
+                    parsedAuthorityOrRawPath = postSchemeParse(addr, p, n, parseAuthority);
+                } else {
+                    int q = scan(addr, p, n, "", "#");
+                    if (q <= p) {
+                        return null;
+                    }    
+                }
+        } else {
+            parsedAuthorityOrRawPath = postSchemeParse(addr, 0, n, parseAuthority);
+        }
+        return new String[] { scheme, parsedAuthorityOrRawPath };
+    }
+
+    private static String postSchemeParse(String addr, int start, int n, boolean parseAuthority) {
+        int p = start;
+        int q = -1;
+        if (at(addr, p, n, '/') && at(addr, p + 1, n, '/')) {
+            p += 2;
+            q = scan(addr, p, n, "", "/");
+            if (q > p) {
+                if (!parseAuthority) {
+                    p = q;
+                }
+            } else if (q >= n) {
+                return null;
+            }    
+        }
+        if (!parseAuthority) {
+            q = scan(addr, p, n, "", "?#");
+        } else if (p == q) {
+            // empty authority should be null
+            return null;
+        }
+        
+        return addr.substring(p, q);
+    }
+    
+    private static boolean at(String addr, int start, int end, char c) {
+        return (start < end) && (addr.charAt(start) == c);
+    }
+    
+    private static int scan(String addr, int start, int end, String err, String stop) {
```

Comment:
I am wondering if should just pass in null for err if there is no err and then just do 

if (err != null && err.indexOf(c) >= 0)